### PR TITLE
Update URL for new py2neo versions

### PR DESCRIPTION
**Pull request by iarspider**

### CODE REVIEW

**Comment by iarspider** [comment]
@adamjstewart please review!

File: var/spack/repos/builtin/packages/py-py2neo/package.py [hunk 1]
```diff
@@ -20,3 +20,12 @@ class PyPy2neo(PythonPackage):
     version('2.0.4', sha256='19074b7b892f2e989f39eae21fc59b26a05e1a820adad8aa58bc470b70d9056d')
 
     depends_on("py-setuptools", type='build')
+
+    def url_for_version(self, version):
+        if version >= Version('2.0.9'):
+            url = 'https://github.com/technige/py2neo'
+        else:
+            url = 'ttps://github.com/nigelsmall/py2neo'
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-py2neo/package.py]: Missing an H. Could we download this package from PyPI instead?

Comment (iarspider) [reply var/spack/repos/builtin/packages/py-py2neo/package.py]: Done

File: var/spack/repos/builtin/packages/py-py2neo/package.py [hunk 2]
```diff
@@ -11,12 +11,13 @@ class PyPy2neo(PythonPackage):
...
-    version('2.0.5', sha256='024b42261b06e5e2c92a1f24e62398847f090862005add0b5c69a79a7e1e87b5')
-    version('2.0.4', sha256='19074b7b892f2e989f39eae21fc59b26a05e1a820adad8aa58bc470b70d9056d')
+    version('4.3.0', sha256='a218ccb4b636e3850faa6b74ebad80f00600217172a57f745cf223d38a219222')
+    version('2.0.8', sha256='06167f5a91a0d9b9b73431baacd876f2d507650a681fdce1fcf3b383a9b991c1')
+    version('2.0.7', sha256='9b154053eb93c7f5fb3ebd48b6a5b99df450d3f2e9c6682153c6f8d59369378c')
+    version('2.0.6', sha256='6bb828d6d3e48b4d095b3f7d79dbb690a47633f0a9812eb62f141b042bab3186')
+    version('2.0.5', sha256='2c04d4223d2d356c4800c586f30c048757334f9391553c852c29aebf2368d101')
+    version('2.0.4', sha256='727726b87268ca1e929191b960a5473409e5bd81559ee83a304951104bb6b866')
 
     depends_on("py-setuptools", type='build')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-py2neo/package.py]: Version 4.3.0 adds many new dependencies:
```python
    "install_requires": [                                                       
        "certifi",                                                              
        "click==7.0",                                                           
        "colorama",                                                             
        "neobolt~=1.7.12",                                                      
        "neotime~=1.7.4",                                                       
        "prompt_toolkit~=2.0.7",                                                
        "pygments~=2.3.1",                                                      
        "pytz",                                                                 
        "urllib3<1.25,>=1.23",                                                  
    ],              
```

**Pull request merged**